factorio-plugin/require.lua

```diff
@@ -13,10 +13,7 @@ local function replace(_, text, diffs)
     local original_name = name
 
     ---Convert the mod name prefix if there is one
-    ---@param match string
-    name = name:gsub("^__(.-)__", function(match)
-      return match
-    end)
+    name = name:gsub("^__(.-)__", "%1", 1)
```

Comment:
lol, I wonder how that happened. This is either some technical debt or I was incredibly unaware of how gsub and patterns :D.
Oh also that 3rd parameter for gsub is optional and I prefer it being omitted if it's the default. Less code to read and comprehend

Reply:
I totally missed that in this case it isn't needed since the ^ anchor is only going to limit it to 1 anyway

Reply:
Ok after playing around with it some more. The modname identifier might have to be dropped in full.

Asumming `mods -> ModA` and `mods -> ModB` and runtime.path of `?.lua` and path strict true

If the workspace.library setting for ModA contains the root path `"/path/to/mods/" then ModA and ModB get loaded twice in sumneko giving duplicate annotation errors however `require "__ModB__" works

If the workspace.library setting for ModA contains `"/path/to/mods/ModB/"` then require doesn't work as it is looking for `/path/to/mods/ModB/ModB/file.lua`
Adding either `../?.lua` or `/path/to/mods/?.lua` to runtime.path does not work either.  This is possibly a bug or limitation in sumneko?

However if the require plugin does `__ModB__/file` -> `file` then it is correctly found.  This comes with the slight drawback that if both `__ModA__` and `__ModB__` have `file.lua` it needs to pointed in the right direction with an @as or @type on the require line.

Reply:
the strict path feature is broken with libraries - meaning it just doesn't find any of them when using require even without the plugin - and using a fully qualified path in `Lua.runtime.path` also simply doesn't work. This is my experience from a while ago and I could be remembering it wrong, but based on raiguard's experience with the strict path setting this still seems to be pretty accurte. I'd definitely call it a bug with `sumneko.lua` which has lead me to not use the strict path setting for the modding workspace